### PR TITLE
updating links from fwlink to real link if exists

### PR DESCRIPTION
**Pull request by eddynaka**
updating links from fwlink to real link if exists - contributes to #3391

### CODE REVIEW

**Comment by mairaw** [comment]
Build seemed stuck

File: docs/framework/wcf/feedback-and-community.md [hunk 1]
```diff
@@ -9,4 +9,4 @@ ms.assetid: ba0fd32b-cbcb-4f5b-b913-d98514ab5650
 # Feedback and Community
 We appreciate your comments and concerns about the Windows Communication Foundation (WCF) or CardSpace documentation. On the bottom of every page in the SDK is the sentence "Send comments about this topic to Microsoft." Click the "comments" link to send your views or suggestions about the specific topic.  
   
- If you are looking for more information, or want to share techniques with others who are interested in WCF, try the [MSDN Windows Communication Foundation forum](https://go.microsoft.com/fwlink/?LinkId=186463).
+ If you are looking for more information, or want to share techniques with others who are interested in WCF, try the [MSDN Windows Communication Foundation forum](https://social.msdn.microsoft.com/Forums/vstudio/en-US/home?forum=wcf).
```
Comment (mairaw) [inline docs/framework/wcf/feedback-and-community.md]: can you undo this one? I'm actually going to delete this article

```suggestion
 If you are looking for more information, or want to share techniques with others who are interested in WCF, try the [MSDN Windows Communication Foundation forum](https://go.microsoft.com/fwlink/?LinkId=186463).
```

File: docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ In Windows Communication Foundation (WCF) applications, a service operation can
...
 ### Implement a service operation asynchronously  
   
-1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](https://go.microsoft.com/fwlink/?LinkId=248221).  
+1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](https://docs.microsoft.com/dotnet/standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-eap).
```
Comment (mairaw) [inline docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md]:
```suggestion
1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](../../dotnet/standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-eap.md).  
```

File: docs/framework/wcf/synchronous-and-asynchronous-operations.md [hunk 3]
```diff
@@ -101,7 +101,7 @@ public class AsyncExample
 }  
 ```  
   
- For more information about the Event-based Asynchronous Pattern, see [The Event-Based Asynchronous Pattern](https://go.microsoft.com/fwlink/?LinkId=232515).  
+ For more information about the Event-based Asynchronous Pattern, see [The Event-Based Asynchronous Pattern](https://docs.microsoft.com/dotnet/standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-overview).
```
Comment (mairaw) [inline docs/framework/wcf/synchronous-and-asynchronous-operations.md]:
```suggestion
 For more information about the Event-based Asynchronous Pattern, see [The Event-Based Asynchronous Pattern](../../standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-overview.md).  
```

File: docs/framework/wcf/troubleshooting-setup-issues.md [hunk 4]
```diff
@@ -46,7 +46,7 @@ This topic describes how to troubleshoot Windows Communication Foundation (WCF)
   
  The following steps must be followed to resolve the problem described previously.  
   
-1. Run [the WMI Diagnosis Utility, version 2.0](https://go.microsoft.com/fwlink/?LinkId=94685) to repair the WMI service. For more information about using this tool, see the [WMI Diagnosis Utility](https://go.microsoft.com/fwlink/?LinkId=94686) topic.  
+1. Run [the WMI Diagnosis Utility, version 2.0](https://go.microsoft.com/fwlink/?LinkId=94685) to repair the WMI service. For more information about using this tool, see the [WMI Diagnosis Utility](https://docs.microsoft.com/) topic.
```
Comment (mairaw) [inline docs/framework/wcf/troubleshooting-setup-issues.md]:
```suggestion
1. Run [the WMI Diagnosis Utility, version 2.0](https://go.microsoft.com/fwlink/?LinkId=94685) to repair the WMI service. For more information about using this tool, see the [WMI Diagnosis Utility](https://docs.microsoft.com/previous-versions/tn-archive/ff404265(v%3dmsdn.10)) article.  
```

File: docs/framework/wcf/wcf-troubleshooting-quickstart.md [hunk 5]
```diff
@@ -130,9 +130,9 @@ public class MyServiceHost : ServiceHost
   
 - [Debugging Windows Authentication Errors](./feature-details/debugging-windows-authentication-errors.md)  
   
-- [Registering Kerberos Service Principal Names by Using Http.sys](https://go.microsoft.com/fwlink/?LinkId=86943)  
+- [Registering Kerberos Service Principal Names by Using Http.sys](https://docs.microsoft.com/previous-versions/sql/sql-server-2008-r2/ms178119(v=sql.105))  
   
-- [Kerberos Explained](https://go.microsoft.com/fwlink/?LinkId=86946)  
+- [Kerberos Explained](https://docs.microsoft.com/)
```
Comment (mairaw) [inline docs/framework/wcf/wcf-troubleshooting-quickstart.md]:
```suggestion
- [Kerberos Explained](https://docs.microsoft.com/)  
```
```suggestion
- [Kerberos Explained](https://docs.microsoft.com/previous-versions/windows/it-pro/windows-2000-server/bb742516(v%3dtechnet.10))  
```

File: docs/framework/wcf/ws-atomictransaction-configuration-utility-wsatconfig-exe.md [hunk 6]
```diff
@@ -19,7 +19,7 @@ wsatConfig [Options]
   
  %SystemRoot%\Microsoft.Net\Framework\v3.0\Windows Communication Foundation\wsatConfig.exe  
   
- If you are running [!INCLUDE[wxp](../../../includes/wxp-md.md)] or [!INCLUDE[ws2003](../../../includes/ws2003-md.md)], you must download an update before running WsatConfig.exe. For more information about this update, see [Update for Commerce Server 2007 (KB912817)](https://go.microsoft.com/fwlink/?LinkId=95340) and [Availability of Windows XP COM+ Hotfix Rollup Package 13](https://go.microsoft.com/fwlink/?LinkId=95341).  
+ If you are running [!INCLUDE[wxp](../../../includes/wxp-md.md)] or [!INCLUDE[ws2003](../../../includes/ws2003-md.md)], you must download an update before running WsatConfig.exe. For more information about this update, see [Update for Commerce Server 2007 (KB912817)](https://go.microsoft.com/fwlink/?LinkId=95340) and [Availability of Windows XP COM+ Hotfix Rollup Package 13](https://support.microsoft.com/help/912817).
```
Comment (mairaw) [inline docs/framework/wcf/ws-atomictransaction-configuration-utility-wsatconfig-exe.md]:
```suggestion
 If you are running [!INCLUDE[wxp](../../../includes/wxp-md.md)] or [!INCLUDE[ws2003](../../../includes/ws2003-md.md)], you must download an update before running WsatConfig.exe. For more information about this update, see [Update for Windows Communication Foundation (KB912817)](https://www.microsoft.com/en-us/download/details.aspx?id=21520).  
```

File: docs/framework/whats-new/index.md [hunk 7]
```diff
@@ -41,13 +41,13 @@ This article does not provide comprehensive information about each new feature a
 
 You can download .NET Framework 4.8  from the following locations:
 
-- [.NET Framework 4.8 Web Installer](https://go.microsoft.com/fwlink/?LinkId=2085155)
+- [.NET Framework 4.8 Web Installer](https://support.microsoft.com/help/912817)
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: This is not where they're going. These are downloads. Don't change then.

File: docs/framework/whats-new/index.md [hunk 8]
```diff
@@ -2054,15 +2054,15 @@ In the .NET Framework 4.5, the following features have been added to make it sim
 
 - Support for WCF services that use internationalized domain names (IDNs).
 
-For more information, see [What's New in Windows Communication Foundation](https://go.microsoft.com/fwlink/?LinkId=228173).
+For more information, see [What's New in Windows Communication Foundation](https://docs.microsoft.com/dotnet/framework/wcf/whats-new).
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]:
```suggestion
For more information, see [What's New in Windows Communication Foundation](../wcf/whats-new.md).
```

File: docs/framework/whats-new/index.md [hunk 9]
```diff
@@ -2108,13 +2108,13 @@ In the .NET Framework 4.5, several new features were added to Windows Workflow F
 
 - Contract-first workflow service development, which provides support for automatically generating activities to match an existing service contract.
 
-For more information, see [What's New in Windows Workflow Foundation](https://go.microsoft.com/fwlink/?LinkId=228176).
+For more information, see [What's New in Windows Workflow Foundation](https://docs.microsoft.com/dotnet/framework/windows-workflow-foundation/whats-new-in-wf-in-dotnet).
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]:
```suggestion
For more information, see [What's New in Windows Workflow Foundation](../windows-workflow-foundation/whats-new-in-wf-in-dotnet.md).
```

File: docs/framework/whats-new/obsolete-types.md [hunk 10]
```diff
@@ -287,198 +287,198 @@ The following tables list the types that have been declared obsolete in system a
...
-|<xref:System.Web.UI.MobileControls.Adapters.XhtmlAdapters.XhtmlValidationSummaryAdapter?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](https://go.microsoft.com/fwlink/?LinkId=157231).|
-|<xref:System.Web.UI.MobileControls.Adapters.XhtmlAdapters.XhtmlValidatorAdapter?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](https://go.microsoft.com/fwlink/?LinkId=157231).|
+|<xref:System.Web.Mobile.CookielessData?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](https://docs.microsoft.com/aspnet/mobile/overview).|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-types.md]: Replace all these

```suggestion
|<xref:System.Web.Mobile.CookielessData?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](/aspnet/mobile/overview).|
```

File: docs/framework/windows-workflow-foundation/authoring-workflows-activities-and-expressions-using-imperative-code.md [hunk 11]
```diff
@@ -23,7 +23,7 @@ A workflow definition is a tree of configured activity objects. This tree of act
   
  [!code-csharp[CFX_WorkflowApplicationExample#49](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#49)]  
   
- For more information about object initializers, see [How to: Initialize Objects without Calling a Constructor (C# Programming Guide)](https://go.microsoft.com/fwlink/?LinkId=161015) and [How to: Declare an Object by Using an Object Initializer](https://go.microsoft.com/fwlink/?LinkId=161016).  
+ For more information about object initializers, see [How to: Initialize Objects without Calling a Constructor (C# Programming Guide)](https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer) and [How to: Declare an Object by Using an Object Initializer](https://docs.microsoft.com/dotnet/visual-basic/programming-guide/language-features/objects-and-classes/how-to-declare-an-object-by-using-an-object-initializer).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/authoring-workflows-activities-and-expressions-using-imperative-code.md]:
```suggestion
 For more information about object initializers, see [How to: Initialize Objects without Calling a Constructor (C# Programming Guide)](../../csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer.md) and [How to: Declare an Object by Using an Object Initializer](../../visual-basic/programming-guide/language-features/objects-and-classes/how-to-declare-an-object-by-using-an-object-initializer.md).  
```

File: docs/framework/windows-workflow-foundation/authoring-workflows-activities-and-expressions-using-imperative-code.md [hunk 12]
```diff
@@ -55,7 +55,7 @@ new Assign<int>
   
  [!code-csharp[CFX_WorkflowApplicationExample#52](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#52)]  
   
- For more information about lambda expressions, see [Lambda Expressions (C# Programming Guide)](https://go.microsoft.com/fwlink/?LinkID=152436) or [Lambda Expressions (Visual Basic)](https://go.microsoft.com/fwlink/?LinkID=152437).  
+ For more information about lambda expressions, see [Lambda Expressions (C# Programming Guide)](https://docs.microsoft.com/dotnet/csharp/programming-guide/statements-expressions-operators/lambda-expressions) or [Lambda Expressions (Visual Basic)](https://docs.microsoft.com/dotnet/visual-basic/programming-guide/language-features/procedures/lambda-expressions).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/authoring-workflows-activities-and-expressions-using-imperative-code.md]:
```suggestion
 For more information about lambda expressions, see [Lambda Expressions (C# Programming Guide)](../../csharp/programming-guide/statements-expressions-operators/lambda-expressions.md) or [Lambda Expressions (Visual Basic)](../../visual-basic/programming-guide/language-features/procedures/lambda-expressions.md).  
```

File: docs/framework/windows-workflow-foundation/configuring-tracking-for-a-workflow.md [hunk 13]
```diff
@@ -129,7 +129,7 @@ if (null != workflowServiceHost)
 ```
 
 > [!NOTE]
-> For more information on tracking profiles, refer to [Tracking Profiles](https://go.microsoft.com/fwlink/?LinkId=201310).
+> For more information on tracking profiles, refer to [Tracking Profiles](https://docs.microsoft.com/dotnet/framework/windows-workflow-foundation/tracking-profiles).
```
Comment (mairaw) [inline docs/framework/windows-workflow-foundation/configuring-tracking-for-a-workflow.md]:
```suggestion
> For more information on tracking profiles, refer to [Tracking Profiles](tracking-profiles.md).
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @eddynaka. Left a few suggestions for you.

**Comment by eddynaka** [comment]
@mairaw just updated those files as you suggested!

File: docs/framework/whats-new/obsolete-types.md [hunk 14]
```diff
@@ -287,198 +287,198 @@ The following tables list the types that have been declared obsolete in system a
...
-|<xref:System.Web.UI.MobileControls.Adapters.XhtmlAdapters.XhtmlValidatorAdapter?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](https://go.microsoft.com/fwlink/?LinkId=157231).|
+|<xref:System.Web.Mobile.CookielessData?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](/aspnet/mobile/overview).|
+|<xref:System.Web.Mobile.DeviceFilterElement?displayProperty=nameWithType>|The System.Web.Mobile.dll assembly has been deprecated and should no longer be used. For information about how to develop ASP.NET mobile applications, see [ASP.NET for Mobiles](https://docs.microsoft.com/aspnet/mobile/overview).|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-types.md]: Can you do a search and replace on this article to remove the domain from the aspnet links?

Comment (eddynaka) [reply docs/framework/whats-new/obsolete-types.md]: done! should i add this rule like /windows?

**Comment by mairaw** [comment]
I've resolved some merge conflicts.

File: docs/framework/whats-new/index.md [hunk 15]
```diff
@@ -795,7 +795,7 @@ The .NET Framework 4.6.2 includes new features in the following areas:
 
 - [Debugging improvements](#Debug462)
 
-For a list of new APIs added to .NET Framework 4.6.2, see [.NET Framework 4.6.2 API Changes](https://github.com/Microsoft/dotnet/blob/master/releases/net462/dotnet462-api-changes.md) on GitHub. For a list of feature improvements and bug fixes in .NET Framework 4.6.2, see [.NET Framework 4.6.2 List of Changes](https://go.microsoft.com/fwlink/?LinkId=708778) on GitHub.  For additional information, see [Announcing .NET Framework 4.6.2](https://devblogs.microsoft.com/dotnet/announcing-net-framework-4-6-2/) in the .NET blog.
+For a list of new APIs added to .NET Framework 4.6.2, see [.NET Framework 4.6.2 API Changes](https://github.com/Microsoft/dotnet/blob/master/releases/net462/dotnet462-api-changes.md) on GitHub. For a list of feature improvements and bug fixes in .NET Framework 4.6.2, see [.NET Framework 4.6.2 List of Changes](https://github.com/Microsoft/dotnet/blob/master/releases/net462/dotnet462-changes.md) on GitHub.  For additional information, see [Announcing .NET Framework 4.6.2](https://devblogs.microsoft.com/dotnet/announcing-net-framework-4-6-2/) in the .NET blog.
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: remove extra space 

```suggestion
For a list of new APIs added to .NET Framework 4.6.2, see [.NET Framework 4.6.2 API Changes](https://github.com/Microsoft/dotnet/blob/master/releases/net462/dotnet462-api-changes.md) on GitHub. For a list of feature improvements and bug fixes in .NET Framework 4.6.2, see [.NET Framework 4.6.2 List of Changes](https://github.com/Microsoft/dotnet/blob/master/releases/net462/dotnet462-changes.md) on GitHub. For additional information, see [Announcing .NET Framework 4.6.2](https://devblogs.microsoft.com/dotnet/announcing-net-framework-4-6-2/) in the .NET blog.
```

File: docs/framework/whats-new/index.md [hunk 16]
```diff
@@ -2108,13 +2108,13 @@ In the .NET Framework 4.5, several new features were added to Windows Workflow F
...
 
-For more information, see [What's New in Windows Workflow Foundation](https://go.microsoft.com/fwlink/?LinkId=228176).
+For more information, see [What's New in Windows Workflow Foundation](../windows-workflow-foundation/whats-new-in-wf-in-dotnet.md).
 
 <a name="tailored" />
 
 ### .NET for Windows 8.x Store apps
 
-Windows 8.x Store apps are designed for specific form factors and leverage the power of the Windows operating system. A subset of the .NET Framework 4.5 or 4.5.1 is available for building Windows 8.x Store apps for Windows by using C# or Visual Basic. This subset is called .NET for Windows 8.x Store apps and is discussed in an [overview](https://go.microsoft.com/fwlink/?LinkId=228491) in the Windows Dev Center.
+Windows 8.x Store apps are designed for specific form factors and leverage the power of the Windows operating system. A subset of the .NET Framework 4.5 or 4.5.1 is available for building Windows 8.x Store apps for Windows by using C# or Visual Basic. This subset is called .NET for Windows 8.x Store apps and is discussed in an [overview](https://docs.microsoft.com/previous-versions/windows/apps/br230302(v=vs.140)) in the Windows Dev Center.
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]:
```suggestion
Windows 8.x Store apps are designed for specific form factors and leverage the power of the Windows operating system. A subset of the .NET Framework 4.5 or 4.5.1 is available for building Windows 8.x Store apps for Windows by using C# or Visual Basic. This subset is called .NET for Windows 8.x Store apps and is discussed in an [overview](https://docs.microsoft.com/previous-versions/windows/apps/br230302(v=vs.140)).
```

**Review by mairaw — COMMENTED** [review]
A few additional comments before merging.

File: docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md [hunk 17]
```diff
@@ -11,7 +11,7 @@ In Windows Communication Foundation (WCF) applications, a service operation can
   
 ### Implement a service operation asynchronously  
   
-1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](https://go.microsoft.com/fwlink/?LinkId=248221).  
+1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](../../dotnet/standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-eap.md).
```
Comment (mairaw) [inline docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md]:
```suggestion
1. In your service contract, declare an asynchronous method pair according to the .NET asynchronous design guidelines. The `Begin` method takes a parameter, a callback object, and a state object, and returns a <xref:System.IAsyncResult?displayProperty=nameWithType> and a matching `End` method that takes a <xref:System.IAsyncResult?displayProperty=nameWithType> and returns the return value. For more information about asynchronous calls, see [Asynchronous Programming Design Patterns](../../standard/asynchronous-programming-patterns/event-based-asynchronous-pattern-eap.md).  
```

**Review by mairaw — APPROVED** [review]
Thanks @eddynaka!

**Pull request merged**